### PR TITLE
Make cockroachDB GA

### PR DESCRIPTION
**Pull request by ritalwar**
- Enhancement

## What does this PR do?

This PR make CockroachDB release GA

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

Run the cockroachDB instance.
Add the integration and metrics should start flowing through Kibana.

## Related issues

- Closes https://github.com/elastic/integrations/issues/3550
- Closes https://github.com/elastic/integrations/issues/480

### CODE REVIEW

**Comment by ishleenk17** [comment]
ecs.yml and base-fields.yml is missing. Please add those

**Comment by ishleenk17** [comment]
I think there is some issue with the basic structuring of the integration. 
I don't see README under _dev/build.
That is used to generate the main rEADME. How are we getting the main README currently ?

**Comment by ritalwar** [comment]
> I think there is some issue with the basic structuring of the integration. I don't see README under _dev/build. That is used to generate the main rEADME. How are we getting the main README currently ?

It is under _dev/build/docs, Please see : https://github.com/ritalwar/integrations/blob/migrate_cockroachdb_ga_3550/packages/cockroachdb/_dev/build/docs/README.md and this is being used to generate main README.

File: packages/cockroachdb/data_stream/status/agent/stream/stream.yml.hbs [hunk 1]
```diff
@@ -5,9 +5,19 @@ hosts:
     - {{this}}
 {{/each}}
 metrics_filters.exclude:
-{{#each metrics_filters.exclude}}
-    - {{this}}
-{{/each}}
+    - raft_process_handleready_latency
+    - raft_scheduler_latency
```
[the remark this answers is not in the document]
Comment (ritalwar) [reply packages/cockroachdb/data_stream/status/agent/stream/stream.yml.hbs:9]: Yes, once dynamic template mapping issue is resolved , will remove these from here.

**Comment by ishleenk17** [comment]
> > I think there is some issue with the basic structuring of the integration. I don't see README under _dev/build. That is used to generate the main rEADME. How are we getting the main README currently ?
> 
> It is under _dev/build/docs, Please see : https://github.com/ritalwar/integrations/blob/migrate_cockroachdb_ga_3550/packages/cockroachdb/_dev/build/docs/README.md and this is being used to generate main README.

My bad, I thought it to be a new integration. Hence was expecting all the files.

**Comment by ritalwar** [comment]
> ecs.yml and base-fields.yml is missing. Please add those

It is already there, https://github.com/ritalwar/integrations/tree/migrate_cockroachdb_ga_3550/packages/cockroachdb/data_stream/status/fields

**Comment by ishleenk17** [comment]
1. Does this work if no username/pwd is provided to the db ? As I don't see any default value for it. 

2. We are doing a * mapping for the fields. As you informed there are around 950 metrics that get exported. 
        Are we making all of these available for user ? The sample json file has a subset of these.

**Comment by ishleenk17** [comment]
Have we tested the integrations with bearer token and ssl.certificate_authorities input ?
Since that is available as a user input.

File: packages/cockroachdb/_dev/deploy/docker/docker-compose.yml [hunk 2]
```diff
@@ -1,10 +1,7 @@
...
+version: '3.2'
 services:
   cockroachdb:
-    image: docker.elastic.co/integrations-ci/beats-cockroachdb:${COCKROACHDB_VERSION:-19.1.1}-1
-    build:
-      context: ./_meta
-      args:
-        COCKROACHDB_VERSION: ${COCKROACHDB_VERSION:-19.1.1}
+    image: cockroachdb/cockroach:${COCKROACHDB_VERSION:-cockroachDB_22}
+    command: start-single-node --insecure
```
Comment (ishleenk17) [inline packages/cockroachdb/_dev/deploy/docker/docker-compose.yml:5]: Is this something specific for integrations ?
As I don't see it present for beats code.

Comment (ritalwar) [reply packages/cockroachdb/_dev/deploy/docker/docker-compose.yml:5]: This is generic cockroachdb command, nothing specific to integrations. 
The --insecure flag is intended for non-production testing only, skipping all TLS encryption and authentication. If this flag is not set, the --certs-dir flag must point to valid certificates.
There were no existing test cases for this package, so created new docker deployment, hence not same as beats.

**Comment by ritalwar** [comment]
> 1. Does this work if no username/pwd is provided to the db ? As I don't see any default value for it.

Yes, it works without username and pwd also and if DB is secured by username/pwd then we can provide this. These are optional fields hence no default value.

> 2. We are doing a * mapping for the fields. As you informed there are around 950 metrics that get exported.
>    Are we making all of these available for user ? The sample json file has a subset of these.

Yes, all these are available for user.

File: packages/cockroachdb/_dev/deploy/docker/docker-compose.yml (hunk 2 above)
Comment (ishleenk17) [reply packages/cockroachdb/_dev/deploy/docker/docker-compose.yml:5]: We are using docker deployment in beats as well, correct ?
Is this setting not required there for testing ?

**Comment by ishleenk17** [comment]
> Yes, all these are available for user.

The sample json is quite a smaller set. Are those cumulative values of those counters ?

File: packages/cockroachdb/_dev/deploy/docker/docker-compose.yml (hunk 2 above)
Comment (ritalwar) [reply packages/cockroachdb/_dev/deploy/docker/docker-compose.yml:5]: It is there also, if you check Dockerfile in beats, please find the link: https://github.com/ritalwar/beats/blob/220983b76d8090a85ffc409cdeb36ae8d3bc688e/x-pack/metricbeat/module/cockroachdb/_meta/Dockerfile#L8

**Comment by ritalwar** [comment]
> > Yes, all these are available for user.
> 
> The sample json is quite a smaller set. Are those cumulative values of those counters ?

No, I think that's because with the cockroachdb version getting used in system test, with this version these many fields were generated and I reported around 900 fields with latest version of cockroachDb i.e v22.1.6. 
I tried with same version of cockroachDb in system test also and now the field count in sample_event.json is almost same, shall I update this version and commit this sample_event.json file?

**Comment by ishleenk17** [comment]
> > > Yes, all these are available for user.
> > 
> > 
> > The sample json is quite a smaller set. Are those cumulative values of those counters ?
> 
> No, I think that's because of cockroachdb version getting used in system test, with this version these many fields were generated and I reported around 900 fields with latest version of cockroachDb i.e v22.1.6. I tried with same version of cockroachDb in system test also and now the field count in sample_event.json is almost same, shall I update this version and commit this sample_event.json file?

I think verifying this should be sufficient. Having those many fields listed on the UI is not required.

**Comment by ritalwar** [comment]
> Have we tested the integrations with bearer token and ssl.certificate_authorities input ? Since that is available as a user input.

Yes, Testing has been done with both the options now.

**Review by ishleenk17 — APPROVED** [review]
Looks good!

**Pull request merged**